server/container/guice/guice-common/src/main/java/org/apache/james/utils/MailRepositoryProbeImpl.java

```diff
@@ -0,0 +1,42 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.utils;
+
+import javax.inject.Inject;
+
+import org.apache.james.mailrepository.api.MailRepositoryStore;
+
+import com.google.common.collect.Iterators;
+
+public class MailRepositoryProbeImpl implements GuiceProbe {
+
+    private final MailRepositoryStore repositoryStore;
+
+    @Inject
+    public MailRepositoryProbeImpl(MailRepositoryStore repositoryStore) {
+        this.repositoryStore = repositoryStore;
+    }
+
+    public int getRepositrySize(String url) throws Exception {
```

Comment:
I think it deserves some javadoc : what is "size" for a mailrepository ?

Reply:
The count of email?